### ISSUE TITLE
I think there is a bug in subpixel prediction

### ISSUE THREAD

**Issue by CodeRed97** [eric-yyjau/pytorch-superpoint#68]
Thanks for your work and sharing! I think I found a bug while learning your paper and your code:
In utils.losses.py, function 'extract_patch_from_points', it seemed the patch extracted does not take the predicted integer coodinate as the center, but take it as the left-top corner, which doesn't make sense in theory. This may cause the prediction accuracy with subpixel poorer than without subpixel. However, this can be easily corrected with some changes in utils.losses.py, function 'extract_patch_from_points' and 'soft_argmax_2d'.
I hope I didn't make a mistake and this could be helpful. 
Thanks for your contribution again.

**Comment by P66094125** [comment]
[CodeRed97](https://github.com/CodeRed97) Can you show your modified code? thank you .

**Comment by foxkw** [comment]
> Thanks for your work and sharing! I think I found a bug while learning your paper and your code: In utils.losses.py, function 'extract_patch_from_points', it seemed the patch extracted does not take the predicted integer coodinate as the center, but take it as the left-top corner, which doesn't make sense in theory. This may cause the prediction accuracy with subpixel poorer than without subpixel. However, this can be easily corrected with some changes in utils.losses.py, function 'extract_patch_from_points' and 'soft_argmax_2d'.(already tested) I hope I didn't make a mistake and this could be helpful. Thanks for your contribution again.

`res_ext = torch.cat(((1-x_res)*(1-y_res), (1-x_res)*y_res, 
            x_res*(1-y_res), x_res*y_res), dim=0)`
Is the design of this res unreasonable? The corresponding distance between the approximate points should be smaller
